vulture_os/toolkit/api_parser/apex/apex.py

```diff
@@ -141,21 +141,67 @@ def get_logs(self, kind, since):
                 else:
                     break
             next_timestamp = int(content["Data"]["NextPage"]["SinceTime"])
+            next_page_token = int(content["Data"]["NextPage"]["PageToken"])
 
         except Exception as e:
             raise ApexAPIError(e)
-        return True, logs, next_timestamp
+        return True, logs, next_page_token, next_timestamp
+
+    def get_logs_with_page_token(self, kind, page_token):
+
+        url_path = f"/WebApp/api/v1/Logs/{kind}?output_format=CEF&page_token={page_token}"
+        logs = []
+        try:
+            while True:
+                self._connect(url_path)
+                content = self.execute_query(self.apex_server_host + url_path)
+                if data_logs := content["Data"]["Logs"]:
+                    logs.extend(data_logs)
+                if next_url := content["Data"].get("Next"):
+                    url_path = f"/WebApp/api/v1/Logs/{kind}" + next_url[next_url.index("?"):]
+                else:
+                    break
+            next_page_token = int(content["Data"]["NextPage"]["PageToken"])
+
+        except Exception as e:
+            raise ApexAPIError(e)
+        return True, logs, next_page_token
```

Comment:
That could have been a single function with both parameters 'since' and 'page_token' as components to add to the URL